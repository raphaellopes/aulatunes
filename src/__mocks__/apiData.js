export const albums = [
  {
    'im:name': {
      label: 'Love & Hate',
    },
    'im:image': [
      {
        label: 'https://is5-ssl.mzstatic.com/image/thumb/Music124/v4/dc/1f/cb/dc1fcbe6-7ff3-7064-e0da-4ce65ee89efb/00602547859044.rgb.jpg/55x55bb.png',
        attributes: {
          height: '55',
        },
      },
      {
        label: 'https://is2-ssl.mzstatic.com/image/thumb/Music124/v4/dc/1f/cb/dc1fcbe6-7ff3-7064-e0da-4ce65ee89efb/00602547859044.rgb.jpg/60x60bb.png',
        attributes: {
          height: '60',
        },
      },
      {
        label: 'https://is3-ssl.mzstatic.com/image/thumb/Music124/v4/dc/1f/cb/dc1fcbe6-7ff3-7064-e0da-4ce65ee89efb/00602547859044.rgb.jpg/170x170bb.png',
        attributes: {
          height: '170',
        },
      },
    ],
    'im:price': {
      label: '$8.99',
      attributes: {
        amount: '8.99000',
        currency: 'USD',
      },
    },
    id: {
      label: 'https://music.apple.com/us/album/love-hate/1440832200?uo=2',
      attributes: {
        'im:id': '1440832200',
      },
    },
    'im:artist': {
      label: 'Michael Kiwanuka',
      attributes: {
        href: 'https://music.apple.com/us/artist/michael-kiwanuka/433457346?uo=2',
      },
    },
    category: {
      attributes: {
        'im:id': '15',
        term: 'R&B/Soul',
        scheme: 'https://music.apple.com/us/genre/music-r-b-soul/id15?uo=2',
        label: 'R&B/Soul',
      },
    },
  },
  {
    'im:name': {
      label: 'Cypress Grove',
    },
    'im:image': [
      {
        label: 'https://is1-ssl.mzstatic.com/image/thumb/Music123/v4/6d/61/20/6d6120c2-4eb7-8d86-14c9-5a29d26057f8/850007715281.jpg/55x55bb.png',
        attributes: {
          height: '55',
        },
      },
      {
        label: 'https://is3-ssl.mzstatic.com/image/thumb/Music123/v4/6d/61/20/6d6120c2-4eb7-8d86-14c9-5a29d26057f8/850007715281.jpg/60x60bb.png',
        attributes: {
          height: '60',
        },
      },
      {
        label: 'https://is4-ssl.mzstatic.com/image/thumb/Music123/v4/6d/61/20/6d6120c2-4eb7-8d86-14c9-5a29d26057f8/850007715281.jpg/170x170bb.png',
        attributes: {
          height: '170',
        },
      },
    ],
    'im:price': {
      label: '$9.99',
      attributes: {
        amount: '9.99000',
        currency: 'USD',
      },
    },
    id: {
      label: 'https://music.apple.com/us/album/cypress-grove/1474602159?uo=2',
      attributes: {
        'im:id': '1474602159',
      },
    },
    'im:artist': {
      label: 'Jimmy "Duck" Holmes',
      attributes: {
        href: 'https://music.apple.com/us/artist/jimmy-duck-holmes/265026843?uo=2',
      },
    },
    category: {
      attributes: {
        'im:id': '2',
        term: 'Blues',
        scheme: 'https://music.apple.com/us/genre/music-blues/id2?uo=2',
        label: 'Blues',
      },
    },
  },
];

export const songs = [
  {
    'im:name': {
      label: 'Leave The Door Open',
    },
    'im:image': [
      {
        label: 'https://is5-ssl.mzstatic.com/image/thumb/Music114/v4/36/8b/5c/368b5c12-725e-e85d-36be-5287461cc0a2/075679788337.jpg/55x55bb.png',
        attributes: {
          height: '55',
        },
      },
      {
        label: 'https://is4-ssl.mzstatic.com/image/thumb/Music114/v4/36/8b/5c/368b5c12-725e-e85d-36be-5287461cc0a2/075679788337.jpg/60x60bb.png',
        attributes: {
          height: '60',
        },
      },
      {
        label: 'https://is4-ssl.mzstatic.com/image/thumb/Music114/v4/36/8b/5c/368b5c12-725e-e85d-36be-5287461cc0a2/075679788337.jpg/170x170bb.png',
        attributes: {
          height: '170',
        },
      },
    ],
    'im:price': {
      label: '$1.29',
      attributes: {
        amount: '1.29000',
        currency: 'USD',
      },
    },
    id: {
      label: 'https://music.apple.com/us/album/leave-the-door-open/1556096966?i=1556097162&uo=2',
      attributes: {
        'im:id': '1556097162',
      },
    },
    'im:artist': {
      label: 'Bruno Mars, Anderson .Paak & Silk Sonic',
      attributes: {
        href: 'https://music.apple.com/us/artist/bruno-mars/278873078?uo=2',
      },
    },
    category: {
      attributes: {
        'im:id': '15',
        term: 'R&B/Soul',
        scheme: 'https://music.apple.com/us/genre/music-r-b-soul/id15?uo=2',
        label: 'R&B/Soul',
      },
    },
  },
  {
    'im:name': {
      label: "Don't Start Now",
    },
    'im:image': [
      {
        label: 'https://is1-ssl.mzstatic.com/image/thumb/Music124/v4/fc/aa/c9/fcaac9e1-327d-3e61-56f2-8911bc63c9d1/190295322175.jpg/55x55bb.png',
        attributes: {
          height: '55',
        },
      },
      {
        label: 'https://is5-ssl.mzstatic.com/image/thumb/Music124/v4/fc/aa/c9/fcaac9e1-327d-3e61-56f2-8911bc63c9d1/190295322175.jpg/60x60bb.png',
        attributes: {
          height: '60',
        },
      },
      {
        label: 'https://is3-ssl.mzstatic.com/image/thumb/Music124/v4/fc/aa/c9/fcaac9e1-327d-3e61-56f2-8911bc63c9d1/190295322175.jpg/170x170bb.png',
        attributes: {
          height: '170',
        },
      },
    ],
    'im:price': {
      label: '$0.69',
      attributes: {
        amount: '0.69000',
        currency: 'USD',
      },
    },
    id: {
      label: 'https://music.apple.com/us/album/dont-start-now/1484636581?i=1484636829&uo=2',
      attributes: {
        'im:id': '1484636829',
      },
    },
    'im:artist': {
      label: 'Dua Lipa',
      attributes: {
        href: 'https://music.apple.com/us/artist/dua-lipa/1031397873?uo=2',
      },
    },
    category: {
      attributes: {
        'im:id': '14',
        term: 'Pop',
        scheme: 'https://music.apple.com/us/genre/music-pop/id14?uo=2',
        label: 'Pop',
      },
    },
  },
];
