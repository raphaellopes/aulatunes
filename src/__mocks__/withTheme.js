import React from 'react';
import { ThemeProvider } from 'styled-components';

export const withTheme = (Component) => (
  <ThemeProvider theme={{ mode: 'light' }}>
    {Component}
  </ThemeProvider>
);
