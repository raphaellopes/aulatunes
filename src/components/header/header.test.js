import { render } from '@testing-library/react';

import { withTheme } from '../../__mocks__/withTheme';
import { Header } from './index';

describe('components | header', () => {
  test('should render the Header', () => {
    const { container } = render(withTheme(<Header />));
    expect(container).toMatchSnapshot();
  });
});
