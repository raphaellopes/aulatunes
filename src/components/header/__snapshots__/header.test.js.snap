// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components | header should render the Header 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #ffffff;
  box-shadow: 0 1px 4px 0 rgba(49,64,92,.1);
}

.c1 {
  position: fixed;
  top: 0;
  right: 0;
  left: 0;
  z-index: 1;
}

.c2 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  max-width: 960px;
  margin: auto;
  padding: 10px;
}

.c3 {
  max-width: 120px;
}

@media (min-width:768px) {
  .c2 {
    padding: 16px;
  }
}

@media (min-width:768px) {
  .c3 {
    max-width: 200px;
  }
}

<div>
  <header
    class="c0 c1"
    data-testid="header"
  >
    <div
      class="c2"
    >
      <img
        alt="Aula Education"
        class="c3"
        src="logo.svg"
      />
    </div>
  </header>
</div>
`;
